packages/xrpc-server/src/server.ts

```diff
@@ -369,26 +368,26 @@ export class Server {
             for await (const item of items) {
               if (item instanceof Frame) {
                 yield item
-                continue
-              }
-              const type = item?.['$type']
-              if (!check.is(item, schema.map) || typeof type !== 'string') {
-                yield new MessageFrame(item)
-                continue
-              }
-              const split = type.split('#')
-              let t: string
-              if (
-                split.length === 2 &&
-                (split[0] === '' || split[0] === nsid)
+              } else if (
+                item != null &&
+                typeof item === 'object' &&
+                '$type' in item &&
+                typeof item.$type === 'string'
               ) {
-                t = `#${split[1]}`
+                const { $type, ...body } = item
+
+                const type =
+                  // "<nsid>#<type>" -> "#<type>"
+                  $type.charCodeAt(nsid.length) === 35 /* "#" */ &&
+                  $type.startsWith(nsid) &&
+                  !$type.includes('#', nsid.length + 1) // no more "#" afterwards
+                    ? $type.slice(nsid.length) // strip nsid (keep "#")
+                    : $type
+
+                yield new MessageFrame(body, { type })
```

Comment:
This change:
- avoids using zod
- avoids clone + delete (performs all in one single operation)
- optimizes `type` transformation (avoid un-necessary array creation)